src/rest-server/src/models/job.js

```diff
@@ -359,6 +370,53 @@ class Job {
     );
   }
 
+  generateExitInfo(code) {
+    if (!_.isNil(code)) {
+      if (!_.isNil(exitInfoMap[code])) {
+        return exitInfoMap[code];
+      } else {
+        if (code > 0) {
+          return {
+            ...exitInfoMap[positiveFallbackExitCode],
+            code,
+          };
+        } else {
+          return {
+            ...exitInfoMap[negativeFallbackExitCode],
+            code,
+          };
+        }
+      }
+    } else {
+      return null;
+    }
+  }
+
+  extractContainerStderr(diag) {
+    const anchor1 = /ExitCodeException exitCode.*?:/;
+    const anchor2 = /at org\.apache\.hadoop\.util\.Shell\.runCommand/;
+    const match1 = diag.match(anchor1);
+    const match2 = diag.match(anchor2);
+    if (match1 !== null && match2 !== null) {
+      const start = match1.index + match1[0].length;
+      const end = match2.index;
+      return diag.substring(start, end).trim();
+    }
+  }
+
+  extractRuntimeOutput(diag) {
```

Comment:
>extractRuntimeOutput [](start = 2, length = 20)

Please also parse RuntimeOutput according to the schema of Dynamic ExitInfo (Runtime Exit Spec)